async-nats/src/jetstream/stream.rs

```diff
@@ -1262,3 +1232,120 @@ pub struct External {
     #[serde(rename = "deliver", skip_serializing_if = "is_default")]
     pub delivery_prefix: Option<String>,
 }
+
+use std::marker::PhantomData;
+
+#[derive(Debug, Default)]
+pub struct Yes;
+#[derive(Debug, Default)]
+pub struct No;
+
+pub trait ToAssign: Debug {}
+
+impl ToAssign for Yes {}
+impl ToAssign for No {}
+
+#[derive(Debug)]
+pub struct PurgeFuture<SEQUENCE, KEEP>
+where
+    SEQUENCE: ToAssign,
+    KEEP: ToAssign,
+{
+    stream: crate::jetstream::stream::Stream,
```

Comment:
You're in the same scope here?

```suggestion
    stream: Stream,
```